### PR TITLE
intel: adsp: make SRAM power-off configurable

### PR DESCRIPTION
**Pull request by lyakh**
Currently the code suggests, that setting the SRAM disabling mask to 0 skips powering off SRAM, whereas in fact it's the address of the mask variable that's checked for NULL. Make this consistent and let platforms select whether SRAM power down should be selected.

### CODE REVIEW

File: soc/intel/intel_adsp/Kconfig [hunk 1]
```diff
@@ -93,11 +93,11 @@ config ADSP_INIT_HPSRAM
 	  Need to init HP SRAM.
 
 config ADSP_POWER_DOWN_HPSRAM
-	bool
+	bool "Switch off HP SRAM during power down"
```
Comment (marc-hb) [inline soc/intel/intel_adsp/Kconfig:96]: The addition of this help line makes this CONFIG_ "visible" and grants the user access to it, as described in the commit subject.

But then the rest of the changes look like (important?) bug fixes, described in the commit message.

So maybe I'm missing something but to me this really looks like this should be two different commits.

**Review by andyross — APPROVED** [review]

File: soc/intel/intel_adsp/Kconfig (hunk 1 above)
Comment (lyakh) [reply soc/intel/intel_adsp/Kconfig:96]: @marc-hb well, can be either way - they're related, but can be separated too. This Kconfig change is small and simple enough IMHO so I merged them into a single commit

**Review by softwarecki — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]
